include/substrait/common/Io.h

```diff
@@ -0,0 +1,30 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#pragma once
+
+#include <string_view>
+
+#include "absl/status/statusor.h"
+#include "substrait/proto/plan.pb.h"
+
+namespace io::substrait {
+
+enum PlanFileEncoding {
+  kBinary = 0,
+  kJson = 1,
+  kProtoText = 2,
+  kText = 3,
+};
+
+// Loads a Substrait plan consisting of any encoding type from the given file.
+absl::StatusOr<::substrait::proto::Plan> loadPlanWithUnknownEncoding(
+    std::string_view input_filename);
+
+// Writes the provided plan to the specified location with the specified
+// encoding type.
+[[maybe_unused]] absl::Status savePlan(
```

Comment:
Just a nitpick but the lack of symmetry between the names `savePlan` and `loadPlanWithUnknownEncoding` would be nice to avoid.  Could we just call it `loadPlan` and document the fact that it guesses the encoding in the comments?  Do we think it's important this goes in the name?

Reply:
The distinction that we're loading any plan seemed important but I agree the symmetry is off.  I've pushed this into the documentation.